File: openapi/code_samples/java/api_locations_getCounties.java
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/location")
public class LocationController {

    @GetMapping("/getAllCounties")
    public ResponseEntity<PageString> getAllCounties(
            @RequestParam(defaultValue = "0") int pageNumber,
            @RequestParam(defaultValue = "50") int pageSize) {
        // Your implementation here to retrieve the list of counties
        // Simulated data for demonstration purposes
        List<String> counties = Arrays.asList("County 1", "County 2", "County 3");

        // Paginate the results
        List<String> paginatedCounties = counties
                .subList(pageNumber * pageSize, Math.min((pageNumber + 1) * pageSize, counties.size()));

        PageString pageString = new PageString();
        pageString.setItems(paginatedCounties);
        pageString.setPageNumber(pageNumber);
        pageString.setPageSize(pageSize);
        pageString.setTotalItems(counties.size());

        return ResponseEntity.ok(pageString);
    }
}

class PageString {
    private List<String> items;
    private int pageNumber;
    private int pageSize;
    private int totalItems;

    // Getters and setters
}

File: openapi/code_samples/csharp/api_locations_getCounties.cs
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;

[ApiController]
[Route("api/[controller]")]
public class LocationController : ControllerBase
{
    [HttpGet]
    [Route("getAllCounties")]
    [SwaggerOperation(Summary = "Retrieves the list of counties in a pageable manner.")]
    [ProducesResponseType(typeof(PageString), 200)]
    [ProducesResponseType(typeof(Fault), 400)]
    public IActionResult GetAllCounties(int pageNumber = 0, int pageSize = 50)
    {
        // Your implementation here to retrieve the list of counties
        // Simulated data for demonstration purposes
        var counties = new List<string>
        {
            "County 1",
            "County 2",
            "County 3"
            // Add more counties as needed
        };

        // Paginate the results
        var paginatedCounties = counties.Skip(pageNumber * pageSize).Take(pageSize).ToList();

        PageString pageString = new PageString
        {
            Items = paginatedCounties,
            PageNumber = pageNumber,
            PageSize = pageSize,
            TotalItems = counties.Count
        };

        return Ok(pageString);
    }
}

public class PageString
{
    public List<string> Items { get; set; }
    public int PageNumber { get; set; }
    public int PageSize { get; set; }
    public int TotalItems { get; set; }
}
